go-controller/pkg/ovn/udn_isolation.go

```diff
@@ -237,3 +239,165 @@ func (oc *DefaultNetworkController) getUDNOpenPortDbIDs(podNamespacedName string
+		map[libovsdbops.ExternalIDKey]string{
+			libovsdbops.ObjectNameKey: networkName,
+			libovsdbops.NetworkKey:    strconv.Itoa(networkID),
+		})
+}
+
+// BuildAdvertisedNetworkSubnetsDropACL builds the advertised network subnets drop ACL:
+// action match                                                                       priority
+// ------ --------------------------------------------------------------------------- --------
+// drop   "(ip[4|6].src == $<ALL_ADV_SUBNETS> && ip[4|6].dst == $<ALL_ADV_SUBNETS>)"    1050
+func BuildAdvertisedNetworkSubnetsDropACL(advertisedNetworkSubnetsAddressSet addressset.AddressSet) *nbdb.ACL {
+	var dropMatches []string
+	v4AddrSet, v6AddrSet := advertisedNetworkSubnetsAddressSet.GetASHashNames()
+	if v4AddrSet != "" {
+		dropMatches = append(dropMatches, fmt.Sprintf("(ip4.src == $%s && ip4.dst == $%s)", v4AddrSet, v4AddrSet))
+	}
+	if v6AddrSet != "" {
+		dropMatches = append(dropMatches, fmt.Sprintf("(ip6.src == $%s && ip6.dst == $%s)", v6AddrSet, v6AddrSet))
+	}
+
+	dropACL := libovsdbutil.BuildACL(
+		GetAdvertisedNetworkSubnetsDropACLdbIDs(),
+		types.AdvertisedNetworkDenyPriority,
+		strings.Join(dropMatches, " || "),
+		nbdb.ACLActionDrop,
+		nil,
+		libovsdbutil.LportEgressAfterLB)
+	dropACL.Tier = types.PrimaryACLTier
+	return dropACL
+}
+
+// addAdvertisedNetworkIsolation adds advertised network isolation rules to the given node.
+// It adds the following ACLs to the node switch:
+// action match                                                                       priority
+// ------ --------------------------------------------------------------------------- --------
+// pass   "(ip[4|6].src == <UDN_SUBNET> && ip[4|6].dst == <UDN_SUBNET>)"                1100
+// drop   "(ip[4|6].src == $<ALL_ADV_SUBNETS> && ip[4|6].dst == $<ALL_ADV_SUBNETS>)"    1050
+func (bnc *BaseNetworkController) addAdvertisedNetworkIsolation(nodeName string) error {
+	var passMatches, cidrs []string
+	var ops []ovsdb.Operation
+
+	addrSet, err := bnc.addressSetFactory.GetAddressSet(GetAdvertisedNetworkSubnetsAddressSetDBIDs())
+	if err != nil {
+		return fmt.Errorf("failed to get advertised subnets addresset %s for network %s: %w", GetAdvertisedNetworkSubnetsAddressSetDBIDs(), bnc.GetNetworkName(), err)
+	}
+
+	for _, subnet := range bnc.Subnets() {
+		ipPrefix := "ip4"
+		if utilnet.IsIPv6CIDR(subnet.CIDR) {
+			ipPrefix = "ip6"
+		}
+		passMatches = append(passMatches, fmt.Sprintf("(%s.src == %s && %s.dst == %s)", ipPrefix, subnet.CIDR, ipPrefix, subnet.CIDR))
+		cidrs = append(cidrs, subnet.CIDR.String())
+
+	}
+
+	addrOps, err := addrSet.AddAddressesReturnOps(cidrs)
+	if err != nil {
+		return fmt.Errorf("failed to add addresses %q to the %s address set for network %s: %w", cidrs, GetAdvertisedNetworkSubnetsAddressSetDBIDs(), bnc.GetNetworkName(), err)
+	}
+	ops = append(ops, addrOps...)
+
+	if len(passMatches) > 0 {
+		passACL := libovsdbutil.BuildACL(
```

Comment:
its weird only deny has a BuildACL util but pass doesn't hehe (i.e this func is doing both building and applying we could have used one more util) but its a nit we are OK here.
can be improved later